fabtests/component/sock_test.c

```diff
@@ -0,0 +1,253 @@
+#include <stdio.h>
+#include <string.h>  
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>   
+#include <arpa/inet.h>    
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/time.h>
+#include <netdb.h>
+#include <time.h>
+#include <poll.h>
+#include <sys/epoll.h>
+
+#define PORT 8888
+
+size_t server_addr_len;
+
+int get_server_addr(char* server_addr, struct sockaddr_storage *server_sock, int *sock)
+{
+	struct addrinfo *res;
+	struct sockaddr_in *in;
+	struct sockaddr_in6 *in6;
+	int ret;
+	int optval = 1;
+
+	ret = getaddrinfo(server_addr, NULL, NULL, &res);
+	if (ret) {
+		printf("getaddrinfo failed\n");
+		return ret;
+	}
+
+	memcpy(server_sock, res->ai_addr, res->ai_addrlen);
+	server_addr_len = res->ai_addrlen;
+
+	switch (server_sock->ss_family) {
+	case AF_INET:
+		in = (struct sockaddr_in *) server_sock;
+		in->sin_port = PORT;
+		break;
+	case AF_INET6:
+		in6 = (struct sockaddr_in6 *) server_sock;
+		in6->sin6_port = PORT;
+		break;
+	default:
+		printf("Unsupported Address family\n");
+		ret = -1;
+		break;
+	}
+
+	*sock = socket(server_sock->ss_family, SOCK_STREAM, 0);
+	if (sock < 0)
+		return -1;
+
+	ret = setsockopt(*sock, SOL_SOCKET, SO_REUSEADDR, (char *) &optval, sizeof(optval));
+	if (ret) {
+		printf("error setting socket options\n");
+		return ret;
+	}
+
+	ret = bind(*sock, (struct sockaddr *)server_sock, server_addr_len);
+
+	return ret;
+}
+
+int server_create(int connections, struct sockaddr_storage server_sock, int sock, int** clients)

```

Comment:
make static, limit line length.  Rename to server_connect() to align with other fabtests.  Clients can just be a global socket array used by both the server and client executions.